device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/MqttIotHubConnection.java

```diff
@@ -346,35 +346,32 @@ public IotHubStatusCode sendMessage(Message message) throws TransportException
             return IotHubStatusCode.BAD_FORMAT;
         }
 
-        synchronized (this.mqttConnectionStateLock) // cannot simultaneously open/close the connection and send a message
```

Comment:
This layer still uses this lock to prevent simultaneous open/close calls, but doesn't need to prevent simultaneous open calls and message receive handling.

Reply:
So if I am reading this correctly, this lock only locks the open and close operations and not message handling correct?

Reply:
Yep. And that is sufficient. Higher layers in this SDK already do some basic locking to prevent sending messages when the client is not open, so it is safe to remove these particular synchronization blocks

Reply:
In case the terminology is new to you, a synchronized block in Java is equivalent to grabbing and then releasing a semaphore.

```java
public void foo()
{
    this.runSomeThreadSafeOperation();

    synchronized (this.lock)
    {
        this.runSomeThreadUnsafeOperation();
    }
}
```

the method foo can be called by multiple threads without worrying about two threads running "runSomeThreadUnsafeOperation" at the same time